### PR TITLE
Add Support for NousResearch/Hermes-2-Pro-Mistral-7B Function Calling

### PR DESCRIPTION
**Pull request by Fanjia-Yan**
This PR introduces support for [ NousResearch/Hermes-2-Pro-Mistral-7B](https://huggingface.co/NousResearch/Hermes-2-Pro-Mistral-7B). Specifically, it includes:

- The implementation of a function calling system prompt passing in OpenAI compatible JSON function descriptions, as detailed on the Huggingface model page.
- The ability to parse raw Hermes output into a standard {"name": ..., "arguments": ...} dictionary format.
- A refactoring of `oss_handler.py` to specify `test_category` during the VLLM batch generation process. This enables special handling for function JSON parameter types in the java and javascript categories.

This PR will result in changes to the leaderboard by adding an entry for one model. The PR that makes changes to the leaderboard webpage is linked here [PR](https://github.com/ShishirPatil/gorilla/pull/328).

### CODE REVIEW

**Review by HuanzhiMao — APPROVED** [review]
LGTM

**Review by CharlieJCJ — CHANGES_REQUESTED** [review]
@Fanjia-Yan Can you add the model in README table, and add an entry in changelog linked to the current PR? Thanks

File: berkeley-function-call-leaderboard/README.md [hunk 1]
```diff
@@ -110,11 +110,11 @@ To generate leaderboard statistics, there are two steps:
...
 
-If you want to run all evaluation at the same time, you can use `all` as the test category.
+If you want to run all evaluations at the same time, you can use `all` as the test category.
 
-Running proprietary model like GPTs, Claude, Mistral-X will requires an API-Key which can be supplied in `openfunctions_evaluation.py`.
+Running proprietary models like GPTs, Claude, Mistral-X will require an API-Key which can be supplied in `openfunctions_evaluation.py`.
 
-If decided to run OSS model, openfunctions evaluation uses vllm and therefore requires GPU for hosting and inferencing. If you have questions or concerns about evaluating OSS models, please reach out to us in our [discord channel](https://discord.gg/grXXvj9Whz).
+If decided to run OSS model, openfunction evaluation uses vllm and therefore requires GPU for hosting and inferencing. If you have questions or concerns about evaluating OSS models, please reach out to us in our [discord channel](https://discord.gg/grXXvj9Whz).
```
Comment (CharlieJCJ) [inline berkeley-function-call-leaderboard/README.md:117]: should be `openfunctions`, for consistencies with filenames.

File: berkeley-function-call-leaderboard/model_handler/hermes_handler.py [hunk 2]
```diff
@@ -0,0 +1,84 @@
...
+<|im_end|>
+        """
+        return formatted_prompt.format(function=function, pydantic_format=pydantic_format,tool_call_format=tool_call_format,prompt=prompt)
+
+    def inference(
+        self, question_file, test_category, num_gpus, fromat_prompt_func=_format_prompt
+    ):
+        return super().inference(
+            question_file, test_category, num_gpus, fromat_prompt_func
+        )
```
[the remark this answers is not in the document]
Comment (HuanzhiMao) [reply berkeley-function-call-leaderboard/model_handler/hermes_handler.py]: I believe this typo has already been fixed? @CharlieJCJ

File: berkeley-function-call-leaderboard/README.md (hunk 1 above)
Comment (HuanzhiMao) [reply berkeley-function-call-leaderboard/README.md:117]: resolved

**Review by ShishirPatil — APPROVED** [review]
LGTM

**Pull request merged**